### ISSUE TITLE
Extraneous CS0234 IDE reports that command-line doesn't report

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#13817]
``` C#
namespace ConsoleApplication43
{
    class Program
    {
        static void Main(string[] args)
        {
            ConsoleApplication43.
        }
    }
}
```

Expected: 
Error	CS1001	Identifier expected
Error	CS1002	; expected

Actual:
Error	CS1001	Identifier expected
Error	CS1002	; expected
Error	CS0234	The type or namespace name '' does not exist in the namespace 'Project1' (are you missing an assembly reference?)

**Comment by davkean** [comment]
No idea if this is IDE or compiler.

**Comment by Pilchie** [comment]
This is because the compiler stops reporting errors if there is a syntax error, but the IDE pushes through to get semantic errors anyway, in case you are in the middle of something else.  Also, because we offer lightbulbs based on diagnostics, and it would be weird if you stopped getting things like "Add Using" if there was a syntax error in your document.

**Comment by davkean** [comment]
It's by design that you get a "The type or namespace name '' does not exist in the namespace 'Project1' (are you missing an assembly reference?)" - help me understand what the user will do with this?

**Comment by Pilchie** [comment]
Technically I guess it's the analyzers, not the IDE ;-P
